### PR TITLE
Add methods for Token CRUD API endpoints

### PR DESCRIPTION
**Pull request by phamann**
### TL;DR
Adds methods to expose API Token management (list, get, create, delete) endpoints: https://docs.fastly.com/api/auth#tokens.

### Notes:
- Upgrades the [go-vcr](https://github.com/dnaeon/go-vcr) dependency to take advantage of the [`AddFilter`](https://github.com/dnaeon/go-vcr#protecting-sensitive-data) method on the recorder instance to automatically strip API keys from requests. 
- Cleaned token credentials form all test recordings
- Didn't implement the batch endpoint

### CODE REVIEW

File: fastly/token.go [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+	}
+	return t, nil
+}
+
+// DeleteTokenInput is used as input to the DeleteToken function.
+type DeleteTokenInput struct {
+	ID string
+}
+
+// DeleteToken revokes a specific token by its id.
```
Comment (peterbourgon) [inline fastly/token.go]:
```suggestion
// DeleteToken revokes a specific token by its ID.
```

File: fastly/token.go [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+		return err
+	}
+
+	if resp.StatusCode != http.StatusNoContent {
+		return fmt.Errorf("Not Ok")
+	}
+	return nil
+}
+
+// DeleteTokenSelf revokes the token used to authorise the requesr
```
Comment (peterbourgon) [inline fastly/token.go]:
```suggestion
// DeleteTokenSelf revokes the token used to authorise the request.
```

File: fastly/fastly_test.go [hunk 3]
```diff
@@ -31,6 +33,12 @@ func record(t *testing.T, fixture string, f func(*Client)) {
 		}
 	}()
 
+	// Add a filter which removes Fastly-Key header from all recorded requests.
+	r.AddFilter(func(i *cassette.Interaction) error {
+		delete(i.Request.Headers, "Fastly-Key")
+		return nil
+	})
```
Comment (philippschulte) [inline fastly/fastly_test.go]: 👏 Why didn't we implement this earlier 😭

**Comment by phamann** [comment]
@philippschulte thanks for the review, let me know if you want me to merge this and cut a release or if you want to?

**Comment by philippschulte** [comment]
@phamann thanks for fixing the build by updating the dependencies. I am happy to take care of the release! However, there is one more step you'll need to do in order to wrap this PR up. Please run `$ go mod tidy` in order to clean up `go.sum`. This will remove:

```
-github.com/dnaeon/go-vcr v0.0.0-20170218072653-87d4990451a8 h1:9W38WyoOkLTitPbj0ltn6yE4PrZ2WJrxUh06pwdKjCQ=
-github.com/dnaeon/go-vcr v0.0.0-20170218072653-87d4990451a8/go.mod h1:aBB1+wY4s93YsC3HHjMBMrwTj2R9FHDzUr9KyGc8n1E=
```

Thanks in advance.

**Comment by phamann** [comment]
@philippschulte Apologies for the delay, I have now tidied the `go.sum` file.

**Pull request merged**